File: app/src/main/java/com/example/vkwall/mvp/fave/FavePresenter.java
package com.example.vkwall.mvp.fave;

import com.example.vkwall.data.model.Profile.FaveProfile.FieldsFave;
import com.example.vkwall.data.remote.ServicesGenerator;
import com.example.vkwall.data.remote.VkApi;
import com.example.vkwall.mvp.base.BasePresenter;

import retrofit2.Call;
import retrofit2.Callback;
import retrofit2.Response;

public class FavePresenter extends BasePresenter<FaveContract> {
    private VkApi vkApi = ServicesGenerator.INSTANCE.createService(VkApi.class);

    @Override
    public void attachView(FaveContract view) {
        super.attachView(view);
    }

    @Override
    public void detachView() {
        super.detachView();
    }

    public void loadFaveProfile(String access_token, int extended, String fields, String item_type, String v) {
        if (getView() != null) {
            getView().showProgressFave();
            vkApi.getFave(access_token, extended, fields, item_type, v).enqueue(new Callback<FieldsFave>() {
                @Override
                public void onResponse(Call<FieldsFave> call, Response<FieldsFave> response) {
                    if (getView() != null) {
                        getView().hideProgressFave();
                        if (response.isSuccessful()) {
                            getView().showFaveProfile(
                                    response.body().getResponse().getItems(),
                                    response.body().getResponse().getProfiles(),
                                    response.body().getResponse().getGroups()
                            );
                        }
                    }
                }

                @Override
                public void onFailure(Call<FieldsFave> call, Throwable t) {
                    if (getView() != null) {
                        getView().hideProgressFave();
                        getView().noInternetFave();
                    }
                }
            });
        }
    }
}

File: app/src/main/java/com/example/vkwall/mvp/fave/FavePresenter.kt
package com.example.vkwall.mvp.fave

import com.example.vkwall.data.DataManager
import com.example.vkwall.data.model.Profile.FaveProfile.FieldsFave
import com.example.vkwall.data.remote.ServicesGenerator
import com.example.vkwall.data.remote.VkApi
import com.example.vkwall.mvp.base.BasePresenter

import retrofit2.Call
import retrofit2.Callback
import retrofit2.Response

class FavePresenter : BasePresenter<FaveContract> {
    private var dataManager: DataManager? = null

    constructor(dataManager: DataManager?) {
        this.dataManager = dataManager
    }

    private lateinit var fieldsFave: Call<FieldsFave>

    fun loadFaveProfile(access_token: String, extended: Int, fields: String, item_type: String, v: String) {
        if (view != null) {
            view.showProgressFave()
            fieldsFave = dataManager!!.getFaveProfile(access_token, extended, fields, item_type, v)
            fieldsFave.enqueue(object : Callback<FieldsFave> {
                override fun onResponse(call: Call<FieldsFave>, response: Response<FieldsFave>) {
                    if (view != null) {
                        view.hideProgressFave()
                        if (response.isSuccessful) {
                            view.showFaveProfile(
                                    response.body()!!.response.items,
                                    response.body()!!.response.profiles,
                                    response.body()!!.response.groups
                            )
                        }
                    }
                }

                override fun onFailure(call: Call<FieldsFave>, t: Throwable) {
                    if (view != null) {
                        view.hideProgressFave()
                        view.noInternetFave()
                    }
                }

            })
        }
    }
}
